server/src/main/java/io/littlehorse/server/streams/topology/core/processors/MetadataProcessor.java

```diff
@@ -42,6 +52,48 @@ public MetadataProcessor(LHServerConfig config, LHServer server, MetadataCache m
 
     public void init(final ProcessorContext<String, CommandProcessorOutput> ctx) {
         this.ctx = ctx;
+
+        maybeInitializeStartupResources();
+    }
+
+    private void maybeInitializeStartupResources() {
+        BackgroundContext context = new BackgroundContext();
+
+        ClusterScopedStore clusterStore =
+                ClusterScopedStore.newInstance(ctx.getStateStore(ServerTopology.METADATA_STORE), context);
+
+        InitConfigModel oldInitConfigModel =
+                clusterStore.get(InitConfigModel.SERVER_INITIALIZED_KEY, InitConfigModel.class);
+
+        // If server has not been initialized..
+        if (oldInitConfigModel == null) {
+            log.info("Initializing Server...");
```

Comment:
Nit: This is "initializing Cluster" not "server". Why? It happens once per cluster, so not all Servers should do this.